### PR TITLE
feat(execute): Support for contract address stubs

### PR DESCRIPTION
**Pull request by marioevz**
## Address Stubs for Pre-deployed Contracts

When running tests on networks that already have specific contracts deployed (such as mainnet or testnets with pre-deployed contracts), you can use the `--address-stubs` flag to specify these contracts instead of deploying new ones.

Address stubs allow you to map contract labels used in tests to actual addresses where those contracts are already deployed on the network. This is particularly useful for:

- Testing against mainnet with existing contracts (e.g., Uniswap, Compound)
- Using pre-deployed contracts on testnets
- Testing on bloat-net, a network containing pre-existing contracts with extensive storage history
- Avoiding redeployment of large contracts to save gas and time

### Using Address Stubs

You can provide address stubs in several formats:

**JSON string:**
```bash
uv run execute remote --fork=Prague --rpc-endpoint=https://rpc.endpoint.io --rpc-seed-key 0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f --rpc-chain-id 12345 --address-stubs '{"DEPOSIT_CONTRACT": "0x00000000219ab540356cbb839cbe05303d7705fa", "UNISWAP_V3_FACTORY": "0x1F98431c8aD98523631AE4a59f267346ea31F984"}'
```

**JSON file:**
```bash
uv run execute remote --fork=Prague --rpc-endpoint=https://rpc.endpoint.io --rpc-seed-key 0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f --rpc-chain-id 12345 --address-stubs ./contracts.json
```

**YAML file:**
```bash
uv run execute remote --fork=Prague --rpc-endpoint=https://rpc.endpoint.io --rpc-seed-key 0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f --rpc-chain-id 12345 --address-stubs ./contracts.yaml
```

### Address Stubs File Format

**JSON format (contracts.json):**
```json
{
  "DEPOSIT_CONTRACT": "0x00000000219ab540356cbb839cbe05303d7705fa",
  "UNISWAP_V3_FACTORY": "0x1F98431c8aD98523631AE4a59f267346ea31F984",
  "COMPOUND_COMPTROLLER": "0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B"
}
```

**YAML format (contracts.yaml):**
```yaml
DEPOSIT_CONTRACT: 0x00000000219ab540356cbb839cbe05303d7705fa
UNISWAP_V3_FACTORY: 0x1F98431c8aD98523631AE4a59f267346ea31F984
COMPOUND_COMPTROLLER: 0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B
```

### How Address Stubs Work

When a test uses a contract label that matches a key in the address stubs, the test framework will:

1. Use the pre-deployed contract at the specified address instead of deploying a new contract
2. Skip the contract deployment transaction, saving gas and time
3. Use the existing contract's code and state for the test

This is particularly useful when testing interactions with well-known contracts that are expensive to deploy or when you want to test against the actual deployed versions of contracts.

### Bloat-net Testing

Address stubs are especially valuable when testing on **bloat-net**, a specialized network that contains pre-existing contracts with extensive storage history. On bloat-net:

- Contracts have been deployed and used extensively, accumulating large amounts of storage data
- The storage state represents real-world usage patterns with complex data structures
- Redeploying these contracts would lose the valuable historical state and storage bloat

Using address stubs on bloat-net allows you to:
- Test against contracts with realistic storage bloat patterns
- Preserve the complex state that has been built up over time
- Avoid the computational and storage costs of recreating this state
- Test edge cases that only emerge with large, real-world storage datasets

## 🔗 Related Issues or PRs
Implements #1976

## ✅ Checklist
- [x] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx --with=tox-uv tox -e lint,typecheck,spellcheck,markdownlint
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [x] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] All: Set appropriate labels for the changes (only maintainers can apply labels).

### CODE REVIEW

File: src/pytest_plugins/execute/pre_alloc.py [hunk 1]
```diff
@@ -31,9 +40,44 @@
...
+        return self.root[item]
+
+    @model_validator(mode="before")
+    @classmethod
+    def load_contents_from_file(cls, value: Any) -> Any:
+        """
+        Try to load from file if the value resembles a path that ends with .json/.yml and the
+        file exists.
+        """
+        if isinstance(value, str):
```
[the remark this answers is not in the document]
Comment (CPerezz) [reply src/pytest_plugins/execute/pre_alloc.py:69]: Tested this and works. I'd 100% go this route. Otherwise JSON string is not working with the code as it is now.

File: src/pytest_plugins/execute/pre_alloc.py [hunk 2]
```diff
@@ -188,8 +255,8 @@ def deploy_contract(
             memory_expansion_gas_calculator = self._fork.memory_expansion_gas_calculator()
             deploy_gas_limit += memory_expansion_gas_calculator(new_bytes=len(bytes(initcode)))
 
-        assert len(initcode) <= MAX_INITCODE_SIZE, (
-            f"initcode too large {len(initcode)} > {MAX_INITCODE_SIZE}"
+        assert len(initcode) <= self._fork.max_initcode_size(), (
+            f"initcode too large {len(initcode)} > {self._fork.max_initcode_size()}"
```
Comment (CPerezz) [inline src/pytest_plugins/execute/pre_alloc.py:259]: ditto

File: src/pytest_plugins/execute/pre_alloc.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marioevz) [reply src/pytest_plugins/execute/pre_alloc.py:69]: I think I implemented this in the incorrect place, it should be a special/separate class method instead of a before validator.
Will update this!

File: src/pytest_plugins/execute/pre_alloc.py [hunk 3]
```diff
@@ -174,7 +239,9 @@ def deploy_contract(
         )
         code = self.code_pre_processor(code, evm_code_type=evm_code_type)
 
-        assert len(code) <= MAX_BYTECODE_SIZE, f"code too large: {len(code)} > {MAX_BYTECODE_SIZE}"
+        assert len(code) <= self._fork.max_code_size(), (
+            f"code too large: {len(code)} > {self._fork.max_code_size()}"
+        )
```
[the remark this answers is not in the document]
Comment (fselmo) [reply src/pytest_plugins/execute/pre_alloc.py:244]: Something caught along the way, likely. We shouldn't re-define this locally as a constant and instead pull it from the `fork`, where it is appropriately defined within the context of the fork that is being used. Same with the `max_initicode_size()` below.

**Review by fselmo — APPROVED** [review]
lgtm!
